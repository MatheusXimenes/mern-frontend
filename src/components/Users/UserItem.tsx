import React from "react";
import { Link } from "react-router-dom";

import Avatar from "../UIElements/Avatar";
import Card from "../UIElements/Card";
import "./UserItem.css";

type userItemProps = {
  id: number;
  image: string;
  name: string;
  placeCount: number;
};

const UserItem = (props: userItemProps) => {
  return (
    <li className="user-item" data-testid={props.id}>
      <Card className="user-item__content">
        <Link to={`/${props.id}/places`}>
          <div className="user-item__image">
            <Avatar image={props.image} alt={props.name} />
          </div>
          <div className="user-item__info">
            <h2>{props.name}</h2>
            <h3>
              {props.placeCount} {props.placeCount === 1 ? "Place" : "Places"}
            </h3>
          </div>
        </Link>
      </Card>
    </li>
  );
};

export default UserItem;
