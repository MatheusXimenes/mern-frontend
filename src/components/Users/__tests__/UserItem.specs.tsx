import React, { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router, Route } from "react-router-dom";
import UserItem from "../UserItem";

const userItemProps = {
  id: 1,
  image:
    "https://images.pexels.com/photos/6530613/pexels-photo-6530613.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1",
  name: "John Doe",
  placeCount: 3,
};

describe("<UserItem/> Component", () => {
  describe("Render", () => {
    it("Should render Link Element", () => {
      render(
        <Router>
          <Route>
            <UserItem
              id={userItemProps.id}
              image={userItemProps.image}
              name={userItemProps.name}
              placeCount={userItemProps.placeCount}
            />
          </Route>
        </Router>
      );
      const link = screen.getByRole("link");
      expect(link).not.toBeNull();
    });
  });

  describe("Snapshot", () => {
    it("Default Variant", () => {
      const tree = renderer.create(
        <Router>
          <Route>
            <UserItem
              id={userItemProps.id}
              image={userItemProps.image}
              name={userItemProps.name}
              placeCount={userItemProps.placeCount}
            />
          </Route>
        </Router>
      );
      expect(tree).toMatchSnapshot();
    });
  });
});
