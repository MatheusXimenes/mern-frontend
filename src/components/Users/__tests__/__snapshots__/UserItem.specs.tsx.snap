// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserItem/> Component Snapshot Default Variant 1`] = `
<li
  className="user-item"
>
  <div
    className="card user-item__content"
  >
    <a
      href="/1/places"
      onClick={[Function]}
    >
      <div
        className="user-item__image"
      >
        <div
          className="avatar undefined"
        >
          <img
            alt="John Doe"
            src="https://images.pexels.com/photos/6530613/pexels-photo-6530613.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1"
            style={
              Object {
                "height": undefined,
                "width": undefined,
              }
            }
          />
        </div>
      </div>
      <div
        className="user-item__info"
      >
        <h2>
          John Doe
        </h2>
        <h3>
          3
           
          Places
        </h3>
      </div>
    </a>
  </div>
</li>
`;
